### DISCUSSION TITLE
Negative lookahead regex to include subdirectory of excluded directory includes the whole directory

### DISCUSSION THREAD

**Discussion by Void-025** [borgbase/vorta#1493] [FAQ] [answered]
I'm trying to exclude everything in my /mnt/ directory except a specific drive ("HDD") using this exclude pattern:
`re:/mnt/(?!HDD/)`
This doesn't seem to be working, based on the size difference of using the above compared to just excluding the entire mount directory (about 5TB difference, even though "HDD" is only 2TB in capacity), it seems to be including the entire /mnt/ directory.
What's wrong with the exclude pattern I'm using?

**Comment by real-yfprojects** [comment, marked as answer]
> about 5TB difference, even though "HDD" is only 2TB in capacity

The vorta GUI currently doesn't support all the pattern types *borg* supports. That's why the GUI won't show you the correct backup size at the moment. In contrast the backup itself should respect the exclude pattern.

**Reply by Void-025** [reply]
I see, but the only reason I noticed this in the first place was that during my backup, it started listing files that should have been excluded, is that affected by this as well then?

**Reply by real-yfprojects** [reply]
No, you will have to remove the leading slash in your pattern: `mnt/(?!HDD/)` since borg stores the paths relative to `/`. I forgot to mention that.

**Reply by Void-025** [reply]
Oh I see, I'm assuming this means all the other normal exclude patterns I have setup (like: "/proc/") also need to have their leading slashes removed (to: "proc/")?

**Reply by Void-025** [reply]
With `re:mnt/(?!HDD/)` it seems to instead be _excluding_ the entire /mnt/ directory instead of including it, even the /mnt/HDD/ which I want included (this is in the backups themselves, which are now 2TB smaller than they should be)...

**Reply by real-yfprojects** [reply]
That makes sense since `(?!HDD/)` will match the empty string. I would try the regular expression `mnt/(?!HDD/|$)`.

**Reply by Void-025** [reply]
Ah I see, after fiddling around with it a bit I've settled on `re:mnt/(?!HDD|$).*(?!/$)/.*` which seems to work for me, thanks for the help!

**Reply by real-yfprojects** [reply]
> I've settled on `re:mnt/(?!HDD|$).*(?!/$)/.*`

That's a very overly complicated pattern. It is aquivalent to `re:mnt/(?!HDD).*/.+` to begin with. As far as I understand it this will only exclude items in subdirectories (unequal to *HDD*) of `/mnt`, not files in `/mnt` or the subdirectories themselves. 

Rereading the borg docs you should prepend `^` to you pattern which matches the beginning of the path so that a `mnt` folder in `/mnt/HDD` isn't excluded.
